internal/controller/cluster_pki.go

```diff
@@ -329,23 +323,26 @@ func (r *ClusterReconciler) ensureLeafCertificate(
 ) error {
 	var secret v1.Secret
 	err := r.Get(ctx, secretName, &secret)
-	if err == nil {
+	switch {
+	case err == nil:
 		return r.renewAndUpdateCertificate(ctx, caSecret, &secret, altDNSNames)
-	}
-
-	serverSecret, err := generateCertificateFromCA(caSecret, commonName, usage, altDNSNames, secretName)
-	if err != nil {
-		return err
-	}
+	case apierrors.IsNotFound(err):
+		serverSecret, err := generateCertificateFromCA(caSecret, commonName, usage, altDNSNames, secretName)
+		if err != nil {
+			return err
+		}
 
-	utils.SetAsOwnedBy(&serverSecret.ObjectMeta, cluster.ObjectMeta, cluster.TypeMeta)
-	for k, v := range additionalLabels {
-		if serverSecret.Labels == nil {
-			serverSecret.Labels = make(map[string]string)
+		utils.SetAsOwnedBy(&serverSecret.ObjectMeta, cluster.ObjectMeta, cluster.TypeMeta)
+		for k, v := range additionalLabels {
+			if serverSecret.Labels == nil {
+				serverSecret.Labels = make(map[string]string)
+			}
+			serverSecret.Labels[k] = v
 		}
-		serverSecret.Labels[k] = v
+		return r.Create(ctx, serverSecret)
+	default:
+		return err
 	}
-	return r.Create(ctx, serverSecret)
```

Comment:
the only case where this logic should be executed is when we get a not found, in the other case we should bubble up the error

Reply:
I meant to refer the old code alone but I couldn't find a way